team/bundles/org.eclipse.compare/compare/org/eclipse/compare/structuremergeviewer/DiffTreeViewer.java

```diff
@@ -670,6 +673,35 @@ private void internalSetSelection(TreeItem ti, boolean fireOpen) {
 		}
 	}
 
+	private void expandCollapseAction(DoubleClickEvent event) {
+
+		ISelection s = event.getSelection();
+
+		if (!s.isEmpty()) {
+			Object e = getElement(s);
+
+			if (e instanceof DiffNode) {
```

Comment:
Since Java 16 you can shorten this to:

```suggestion
			if (e instanceof DiffNode d) {
```
and then remove the 
```
DiffNode d = (DiffNode) e;
```
line

Reply:
Done